Source/dllmain.cpp

```diff
@@ -0,0 +1,169 @@
+    if (myfile.is_open()) {
+        std::stringstream ss;
+        ss << aMsg->time << ";";
+        ss << aMsg->message << ";";
+        ss << aMsg->hwnd << ";";
+        ss << aMsg->lParam << ";";
+        ss << aMsg->wParam << ";";
+        ss << aMsg->pt.x << ";";
+        ss << aMsg->pt.y << ";";
+        printLog(ss.str());
+    }
+}
+
+BOOL APIENTRY DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved)
+{
+    switch (fdwReason)
+    {
+    case DLL_PROCESS_ATTACH:
+        g_hDll = hinstDLL;
+        //MessageBoxA(0, "Attach now", "Attach now", 0);
+        return true;
+    case DLL_PROCESS_DETACH:
+        UnsetMsgHook();
+        return true;
+    }
+}
+
+BOOL OpenLogFile(char* path) {
+    if (path)
+        myfile.open(path);
+    return myfile.is_open();
+}
+
+void infinityWait() {
+    std::condition_variable cv;
+    std::mutex m;
+    std::unique_lock<std::mutex> lock(m);
+    cv.wait(lock, [] {return false; });
+}
+
+void initSocket() {
+    WSADATA wsaData;
+    WSAStartup(MAKEWORD(2, 2), &wsaData);
+    sock = socket(AF_INET, SOCK_DGRAM, 0);
+    if (sock == INVALID_SOCKET) {
+        return;
+    }
+    BOOL enabled = TRUE;
+    if (setsockopt(sock, SOL_SOCKET, SO_BROADCAST, (char*)&enabled, sizeof(BOOL)) < 0) {
+        return;
+    }
+    Sender_addr.sin_family = AF_INET;
+    Sender_addr.sin_port = htons(PORT);
+    Sender_addr.sin_addr.s_addr = inet_addr("localhost");
+    socketInitialized = true;
+}
+
+/*
+Call this function after dll injected
+*/
+BOOL SetMsgHook(char* path = nullptr)
+{
+    OpenLogFile(path);
+    //MessageBoxA(0, "Attach now", "Attach now", 0);
+    g_hHook = SetWindowsHookEx(WH_GETMESSAGE, SysMsgProc, g_hDll, 0);
+    if (g_hHook != NULL)
+    {
+        printLog("Hook successfully set!");
+        printLog("time;hwnd;lParam;wParam;message;pt.x;pt.y");
+    }
+    else
+    {
+        HRESULT hr = HRESULT_FROM_WIN32(GetLastError());
+        _com_error err(hr);
+        LPCTSTR errMsg = err.ErrorMessage();
+        printLog("Error when hook set");
+    }
+
+    initSocket();
+    infinityWait();
+    return g_hHook != NULL;
+}
+
+BOOL UnsetMsgHook()
+{
+    if (g_hHook == NULL)
+        return false;
+
+    if (myfile.is_open())
+        myfile.close();
+
+    BOOL isUnhooked = UnhookWindowsHookEx(g_hHook);
+    if (isUnhooked)
+    {
+        printLog("Unhook");
+        g_hHook = NULL;
+    }
+
+    return isUnhooked;
+}
+
+LRESULT CALLBACK SysMsgProc(int nCode, WPARAM wParam, LPARAM lParam)
+{
+    if (nCode < 0)
+        return CallNextHookEx(g_hHook, nCode, wParam, lParam);
+
+    LPMSG aMsg = (LPMSG)lParam;
+    if (aMsg) {
+        messageToFile(aMsg);
+        memcpy(&buff, aMsg, sizeof(*aMsg));
+        if (socketInitialized) {
+            if (sendto(sock, buff, sizeof(*aMsg), 0, (sockaddr *)&Sender_addr, sizeof(Sender_addr)) < 0)
+            {
+                closesocket(sock);
+                printLog("Socket closed!");
+                return 0;
```

Comment:
We're breaking hook chain here... but we shouldn't. In theory we should never meet this situation because socket error handling must be before setting the hook. I'd suggest just skipping everything here and do `CallNextHookEx()` in any case.